### PR TITLE
[CMake] Add option to control assertions

### PR DESCRIPTION
**Pull request by TingPing**
#### 1ef24965e556d8cf0e4344b0f5395d7e02cf005f
<pre>
[CMake] Add option to control assertions
<a href="https://bugs.webkit.org/show_bug.cgi?id=259982">https://bugs.webkit.org/show_bug.cgi?id=259982</a>

Reviewed by Don Olmstead.

The macOS port already allows building `release+assert` but the
cmake project didn&apos;t expose this.

This adds a `-DENABLE_ASSERTS` option to enable with a normal
release build.

* Source/JavaScriptCore/API/tests/testapi.c:

Handle ASSERT_ENABLED being already defined here.

* Source/JavaScriptCore/bytecode/CodeBlock.h:
* Source/WTF/wtf/RefCounter.h:
(WTF::RefCounter&lt;T&gt;::Count::deref):
* Source/WebCore/layout/LayoutContext.h:
* Source/WebCore/platform/graphics/nicosia/cairo/NicosiaPaintingContextCairo.h:
* Source/WebCore/platform/mediastream/MediaConstraints.cpp:
(WebCore::FlattenedConstraint::append):
* Source/WebCore/platform/mediastream/MediaConstraints.h:

There are a few cases in the codebase where they mix checking
NDEBUG and ASSERT_ENABLED between the declaration/definition/usage.
This just changes them to be in sync.

* Source/cmake/OptionsCommon.cmake:

Add the `ENABLE_ASSERTS` option.

Canonical link: <a href="https://commits.webkit.org/266778@main">https://commits.webkit.org/266778@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/578eb0b147df319c0b14486d24c69a3dc7e64dee

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14739 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15046 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15398 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16487 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13897 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/14876 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/17567 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15145 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/16546 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14919 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15421 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/12514 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17222 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/14841 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/12705 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/13297 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/20301 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/12624 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13779 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13464 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/16715 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/14011 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14008 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/11850 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/14940 "Built successfully") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/14636 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13304 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/13153 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3823 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17641 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/15170 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1765 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13851 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/3633 "Passed tests") |

### CODE REVIEW

File: Source/JavaScriptCore/b3/air/AirCCallingConvention.cpp [hunk 1]
```diff
@@ -32,6 +32,7 @@
 #include "AirCode.h"
 #include "B3CCallValue.h"
 #include "B3Procedure.h"
+#include "B3ProcedureInlines.h"
```
Comment (donny-dont) [inline Source/JavaScriptCore/b3/air/AirCCallingConvention.cpp]: This change seems unrelated.

**Review by donny-dont — APPROVED** [review]
LGTM minus what looks like an unrelated change

**Comment by mcatanzaro** [comment]
Heh, I've been wanting to do this for years, but assumed it'd be a lot harder to make it build. Nice.
